.buildkite/pipeline.yml

```diff
@@ -17,16 +28,18 @@ steps:
 
         buildkite-agent artifact upload bundle/android/App.js
 
+  - label: "Build Android RN Aztec & Publish to S3"
+    key: "publish-react-native-aztec-android"
+    <<: *publish-android-artifacts-docker-container
+    command: |
+        cd ./gutenberg/packages/react-native-aztec/android
+        ./gradlew -PwillPublishReactNativeAztecBinary=true :prepareToPublishToS3 `prepare_to_publish_to_s3_params` :publish
+        buildkite-agent meta-data set "PUBLISHED_REACT_NATIVE_AZTEC_ANDROID_VERSION" `cat ./build/published-version.txt`
```

Comment:
Actually, I _mostly_ feel the opposite. It's just much easier to see what's going on if everything was in the pipeline - as long as the pipeline file doesn't become a huge mess.

I'll consider making this change in a future PR, but I'd prefer not to do it right now.

Reply:
👍 

I agree that reading it inline is more informative. From a maintenance point of view, though, moving the shell code into a shell script would allow the author's editor to help out.

But the real reason I said it was because I wasn't symmetric with the other step, which is a petty reason to make a change in something that already works 🙃

Reply:
I am not a big fan of the asymmetry either, but I think the reason the asymmetry is this obvious is because we don't have the other Buildkite steps yet.

---

I don't know what I like just yet when it comes to Buildkite configurations, so for now I am just going with whatever feels best. We'll have plenty opportunities to make it all consistent and nice, but I feel the exploration is useful in making that decision.

Having said that, if you feel strongly about having a separate script, I don't mind doing it now.

Reply:
TIL about `buildkite-agent meta-data set` 👍 

nit that [while reading the documentation](https://buildkite.com/docs/agent/v3/cli-meta-data#setting-data-description), they suggest to use indirection / stdin piping – rather than "\`cat \<file\>\`" –, so something like this:

```
buildkite-agent meta-data set "PUBLISHED_REACT_NATIVE_AZTEC_ANDROID_VERSION" << ./build/published-version.txt
```

Using a pipe or indirection has the advantages that (1) if the content of the file contains spaces, it won't mess the command line args like `cat <file>` would (even if in our case it's unlikely, better avoid the risk and prevent injection), and (2) it avoids spawning new shell process (to interpret the commant in the backticks, then one to run the `cat` command). It's probably not a big deal here in this use case of ours, but good to know for the future 🙂

Reply:
It looks like I missed that part of the documentation 🤦